20210530-acme-zhiwei.md

```diff
@@ -0,0 +1,200 @@
+# Add ACME protocol support for certificate management with DNS
+
+| Status        | (Proposed / Accepted / Implemented / Obsolete)       |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | [12](https://github.com/coredns/rfc/pull/12) |
+| **Author(s)** | Zhi Wei Chin(@[chinzhiweiblank](https://github.com/chinzhiweiblank)) |
+| **Sponsor**   | Yong Tang (@[yongtang](https://github.com/yongtang)), Paul Greenberg (@[greenpau](https://github.com/greenpau)) |
+| **Updated**   | 2021-06-05                                       |
+| **Obsoletes** | |
+
+## Objective
+
+This project idea aims to add ACME protocol support for certificate management with DNS.
+
+## Motivation and Scope
+Currently CoreDNS supports DNS over HTTPS, which is a protocol for performing DNS resolution via the HTTPS protocol. However, there is no out-of-box certificate management provided by CoreDNS. Hence it is up to the users to manage the certificate lifecycle.
```

Comment:
I think this can be expanded a little bit to give users not familiar with DNS some background on why DNS is related to certificate management.

There are several reasons DNS is import for certificate management. The biggest reason is that certificate management requires identification of the person for certificate request (or renew).

DNS is a great way to resolve the identification issue: If a person can prove the ownership of a DNS domain name (e.g. myname.com), then this person's identification is clear.

The ownership of DNS domain name is done through a challenge: if you claim you own `myname.com`, can you modify your domain name's DNS record? If the DNS record is modified underneath `myname.com` as requested then we know you own `myname.com`.

Of course, DNS challenge can only be done through DNS server software (or through cloud vendor's DNS service like Route53/etc).

CoreDNS is a DNS server software. 

In CoreDNS case, there are potentially two scenarios:
1. CoreDNS needs a certificate management tool to solve its own DNS over TLS certificate issue.
2. CoreDNS potentially can be used to provide generic certificate management method for scenarios beyond serving DNS over TLS issue. This is not mentioned in the original project page as you can also use HTTP challenge (which many people will use due to the familiarity of http than DNS.).

In this project we will focus on 1) though it would be ideal to at least provide more background information.

Reply:
I updated the RFC's Motivation and Scope with this.  About the DNS Challenge, the ACME protocol requires creating a TXT record and place it on my domain. If I run CoreDNS on my local machine, how does CoreDNS create a TXT record on the domain? And how can the CA find this record on CoreDNS on my local machine?